avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -90,6 +93,109 @@
 
   // Connect -> Avro
 
+  @Test
+  public void testCyclicalAvroSchema() {
+    AvroDataConfig avroDataConfig = new AvroDataConfig.Builder()
+        .with(AvroDataConfig.CONNECT_META_DATA_CONFIG, false)
+        .with(AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, false)
+        .build();
+    AvroData listAvroData = new AvroData(avroDataConfig);
+    String linkedListAvroSchema =
+        "{\"type\": \"record\",\"name\": \"linked_list\",\"fields\" : "
+            + "[{\"name\": \"value\", \"type\": \"long\"},"
+            + "{\"name\": \"next\", \"type\": [\"null\", \"linked_list\"],\"default\" : null}]}";
+    org.apache.avro.Schema.Parser avroParser = new org.apache.avro.Schema.Parser();
+    org.apache.avro.Schema avroSchema = avroParser.parse(linkedListAvroSchema);
+
+    GenericRecord next = new GenericRecordBuilder(avroSchema)
+        .set("value", 2l)
+        .set("next", null)
+        .build();
+
+    GenericRecord headNode = new GenericRecordBuilder(avroSchema)
+        .set("value", 3l)
+        .set("next", next)
+        .build();
+
+    SchemaAndValue schema = listAvroData.toConnectData(avroSchema, headNode);
+    GenericRecord genericRecord = (GenericRecord) listAvroData.fromConnectData(
+        schema.schema(), schema.value());
+
+    assertEquals(headNode, genericRecord);
```

Comment:
Actually here it looks like we're doing roundtrip tests, we should just add a new section for those if this doesn't fall into only either the Avro -> Connect or Connect -> Avro tests.

Also, I think we want to assert more than this. It guarantees the round trip works for Avro -> Connect -> Avro, but doesn't validate Connect -> Avro -> Connect and doesn't validate the expected structure of the Connect version (for which we should be able to assert at least some basics, and some stuff unique to cyclic schemas, such as that the top-level and field schema are equal). Same question for following couple of tests.

Reply:
sure will add some  assertions once we do Avro -> Connect